packages/core/src/components/forms/_file-upload.scss

```diff
@@ -135,3 +118,22 @@ $file-upload-button-width-large: $pt-grid-size * 8.5;
     }
   }
 }
+
+.pt-file-upload-input {
+  @include pt-input();
+  @include overflow-ellipsis();
+  position: absolute;
+  top: 0;
+  right: 0;
+  left: 0;
+  height: $pt-input-height;
+  padding-right: $file-upload-button-width + $input-padding-horizontal;
+  line-height: $pt-input-height;
```

Comment:
`height` and `line-height` are redundant -- they come from `pt-input()` mixin. please audit these properties and remove redundancies (might be others).